crates/fuel-core/src/service/sub_services.rs

```diff
@@ -134,9 +134,9 @@ pub fn init_sub_services(
     };
     let producer_adapter = BlockProducerAdapter::new(block_producer);
 
-    let poa_config: fuel_core_poa::Config = config.try_into()?;
+    let poa_config: fuel_core_poa::Config = config.into();
     let production_enabled =
-        !matches!(poa_config.trigger, Trigger::Never) || config.manual_blocks_enabled;
+        !matches!(poa_config.trigger, Trigger::Never) || config.debug;
```

Comment:
1. It may be useful to have an info message saying that production is enabled or disabled
2. If `debug` turns off block production, what can we do if we need extra debug features to debug block production? This also seems to imply that we shouldn't add debug features related to block production, which isn't obvious.

Reply:
1. Done
2. `debug` only enables production in the case when it should be off. What do you mean by "debug block production"? I don't see how we can do that=)

Reply:
That  makes sense. I misread `production_enabled` as `production_disabled`, and I was confused why debug would turn off block production